### PR TITLE
Make timer only start once and shutdown it for tests

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

File: sdk/src/main/java/ly/count/android/sdk/Countly.java [hunk 1]
```diff
@@ -241,13 +241,8 @@ public static Countly sharedInstance() {
      * Creates a new ConnectionQueue and initializes the session timer.
      */
     Countly() {
-        timerService_ = Executors.newSingleThreadScheduledExecutor();
-        staticInit();
-    }
-
-    private void staticInit() {
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/Countly.java:248]: this function was not needed anymore

File: sdk/src/main/java/ly/count/android/sdk/Countly.java [hunk 2]
```diff
@@ -260,12 +255,7 @@ private void startTimerService(ScheduledExecutorService service, ScheduledFuture
             timerDelay = 1;
         }
 
-        timerFuture = service.scheduleWithFixedDelay(new Runnable() {
-            @Override
-            public void run() {
-                onTimer();
-            }
-        }, timerDelay, timerDelay, TimeUnit.SECONDS);
+        timerFuture = service.scheduleWithFixedDelay(this::onTimer, timerDelay, timerDelay, TimeUnit.SECONDS);
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/Countly.java:258]: java lambda things

File: sdk/src/main/java/ly/count/android/sdk/Countly.java [hunk 3]
```diff
@@ -427,11 +417,13 @@ public synchronized Countly init(CountlyConfig config) {
                 config.sdkInternalLimits.maxStackTraceLineLength = maxStackTraceLineLengthDefault;
             }
 
+            long timerDelay = TIMER_DELAY_IN_SECONDS;
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/Countly.java:420]: for preventing starting the timer twice

File: sdk/src/main/java/ly/count/android/sdk/Countly.java [hunk 4]
```diff
@@ -843,6 +852,7 @@ public synchronized void halt() {
         L.i("Halting Countly!");
         sdkIsInitialised = false;
         L.SetListener(null);
```
Comment (arifBurakDemiray) [inline sdk/src/main/java/ly/count/android/sdk/Countly.java:854]: we stop timer only halt, it is needed for test purposes

**Review by turtledreams — APPROVED** [review]

**Pull request merged**